File: src/commonMain/kotlin/com/scoreo/application/FindGameTypeByIdUseCase.kt
package com.scoreo.application

import com.scoreo.domain.model.GameType
import com.scoreo.domain.port.GameTypeRepository

class FindGameTypeByIdUseCase(private val repository: GameTypeRepository) {
    operator fun invoke(gameTypeId: String): GameType? = repository.findById(gameTypeId)
}

File: apps/scoreo/src/application/findGameTypeByIdUseCase.ts
import type { GameType } from '../domain/model/gameType'
import type { GameTypeRepository } from '../domain/port/gameTypeRepository'

export class FindGameTypeByIdUseCase {
  constructor(private readonly repository: GameTypeRepository) {}

  invoke(gameTypeId: string): GameType | undefined {
    return this.repository.findById(gameTypeId)
  }
}
